### DISCUSSION TITLE
How to build/pip install a local version?

### DISCUSSION THREAD

**Discussion by KrisThielemans** [ETSInitiative/PETSIRD#48] [Q&A] [answered]
I'd like to be able to `pip install` a PETSIRD version that I changed locally. I know very little about Python package distribution, so need some help... I tried
```sh
mkdir -p ~/devel
cd ~/devel
python3 -m venv petsirdenv
source ./petsirdenv/bin/activate
pip install -q build
cd PETSIRD/python
python -m build
```
with the intention to follow that by something like
```
pip install  petsird-*.whl
```
However, the first step leads to
```
FileNotFoundError: [Errno 2] No such file or directory: 'petsird/version.py'
```
@casperdcl @johnstairs can you help, please?

**Comment by casperdcl** [comment, marked as answer]
```sh
cd model
yardl generate
pip install ../python
```

and if you don't want to keep re-installing (i.e. running `yardl generate` won't need another `pip install`):

```sh
pip install -e ../python
```

**Reply by KrisThielemans** [reply]
cool
